packages/peregrine/lib/talons/MagentoRoute/addToCache.js

```diff
@@ -0,0 +1,15 @@
+const addToCache = async (...urls) => {
+    if (!window.caches) {
```

Comment:
It is good to see caching but can we delegate this work to the SW? 2 reasons:

1. No main thread cycles wasted
2. `cacheName` can change in the future and having it hardcoded here can be a potential cause for bugs. We can leave this to the SW and if the `cacheName` changes in the future workbox will take care of it.

You can use [sendMessageToSw](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/util/swUtils.js#L90) to send a message along with the list of URL you want to cache and have a [handler](https://github.com/magento/pwa-studio/blob/develop/packages/venia-concept/src/ServiceWorker/Utilities/messageHandler.js#L27) on the SW do the caching part.

For instance:

On the UI:

```es6
sendMessageToSW('PREFETCH_URLS', { urls: ['https://develop.pwa-venia.com/main.css', 'https://develop.pwa-venia.com/font.woff'] })
```

In SW:

```es6
registerMessageHandler('PREFETCH_URLS', prefetchUrls)
```

Check out this example:

https://github.com/magento/pwa-studio/blob/21aadbc054e9d874f0bdaf86ee346e03801c5e09/packages/peregrine/lib/talons/ProductImageCarousel/useProductImageCarousel.js#L36

https://github.com/magento/pwa-studio/blob/21aadbc054e9d874f0bdaf86ee346e03801c5e09/packages/venia-concept/src/ServiceWorker/Utilities/imageCacheHandler.js#L144

Reply:
Good feedback, I agree with you on all points. I'd like to handle the change in a separate PR, though. This one is just a 1:1 refactor—nothing about `addToCache` has changed here.

Reply:
The `addToCache` implementation is different here, though. The original implementation had some big problems, so I'm not about bringing it back, but I think this isn't populating the cache correctly. Can you confirm that it's actually caching and reusing the `urlResolver` responses? I think this is just caching an app shell instead of a GraphQL response.

Reply:
Call me crazy, but did I do something other than just move [this code](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/Router/magentoRouteHandler.js#L32-L42) into its own file?